rts/Sim/Path/HAPFS/PathDataTypes.h

```diff
@@ -37,9 +37,10 @@ struct PathNode {
 
 
 /// functor to define node priority
+/// This needs to guarantee that the sorting is stable.
 struct lessCost {
 	inline bool operator() (const PathNode* x, const PathNode* y) const {
-		return (x->fCost == y->fCost) ? (x->gCost < y->gCost) : (x->fCost > y->fCost);
+		return std::tie(x->fCost, x->gCost, x->nodeNum) > std::tie(y->fCost, y->gCost, y->nodeNum);
```

Comment:
if `fCost` and `gCost` evaluated in different direction, don't you need to change `std::tie(x->fCost` to `std::tie(y->fCost` and vice versa?

Reply:
Not if we're going to put it this way around. Like I said, I can't think of a reason to evaluate them differently. fCost = gCost + hCost. Most of the time fCost resolves the order, and on a few occassions it will be gCost.

Reply:
Oh, actually I can.

Reply:
Okay. Switched it back. It makes sense that away around. To avoid such confusion again in the future, I've also added a comment.